drivers/overlay/overlay.go

```diff
@@ -1456,6 +1456,32 @@ func (d *Driver) get(id string, disableShifting bool, options graphdriver.MountO
 		return "", err
 	}
 
+	// user namespace requires this to move a directory from lower to upper.
+	rootUID, rootGID, err := idtools.GetRootUIDGID(options.UidMaps, options.GidMaps)
+	if err != nil {
+		return "", err
+	}
+
+	mergedDir := d.getMergedDir(id, dir, inAdditionalStore)
+	// Attempt to create the merged dir only if it doesn't exist.
+	if err := fileutils.Exists(mergedDir); err != nil && os.IsNotExist(err) {
+		if err := idtools.MkdirAllAs(mergedDir, 0o700, rootUID, rootGID); err != nil && !os.IsExist(err) {
```

Comment:
I think we added the outer `fileutils.Exists` because `MkdirAllAs` chowns an existing directory, while in this case we don't want to do it when the merged directory exists (it might be in an additional store).

I'll change it to use `MkdirAllAndChownNew`